Mathlib/Order/OmegaCompletePartialOrder.lean

```diff
@@ -286,17 +305,33 @@ def Continuous' (f : α → β) : Prop :=
   ∃ hf : Monotone f, Continuous ⟨f, hf⟩
 #align omega_complete_partial_order.continuous' OmegaCompletePartialOrder.Continuous'
 
-lemma isLUB_of_scottContinuous {c : Chain α} {f : α → β} (hf : ScottContinuous f) :
-    IsLUB (Set.range (Chain.map c ⟨f, (ScottContinuous.monotone hf)⟩)) (f (ωSup c)) := by
+/-- ωScottContinuous - Scott Continuous over Chains-/
+def ωScottContinuous (f : α → β) := ScottContinuousOn (Set.range fun c : Chain α => Set.range c) f
+
+lemma ωScottContinuous.monotone {f : α → β} (h : ωScottContinuous f) : Monotone f :=
+  ScottContinuousOn.monotone _ (fun a b hab => by
+    use ordered_pair_to_chain a b hab; exact ordered_pair_to_chain_range a b hab) h
+
+lemma ωScottContinuous.isLUB {c : Chain α} {f : α → β} (hf : ωScottContinuous f) :
+    IsLUB (Set.range (Chain.map c ⟨f, (ωScottContinuous.monotone hf)⟩)) (f (ωSup c)) := by
   simp only [map_coe, OrderHom.coe_mk]
   rw [(Set.range_comp f ↑c)]
-  exact hf (Set.range_nonempty ↑c) (IsChain.directedOn (isChain_range c)) (isLUB_range_ωSup c)
+  exact hf (by simp only [Set.mem_range, exists_apply_eq_apply]) (Set.range_nonempty ↑c)
+    (IsChain.directedOn (isChain_range c)) (isLUB_range_ωSup c)
 
-lemma ScottContinuous.continuous' {f : α → β} (hf : ScottContinuous f) : Continuous' f := by
+lemma continuous'_iff_ωScottContinuous {f : α → β} : Continuous' f ↔ ωScottContinuous f := by
```

Comment:
Could you straightaway replace `continuous'` by `ωScottContinuous`?

Reply:
I also think we don't need a separate definition `OmegaCompletePartialOrder.Continuous`. It's just `ωScottContinuous ⇑f`.

Reply:
Probably we should define (I'm guessing):
```
def scottHausdorffOn (α : Type*) [Preorder α] (D : Set (Set α)) : TopologicalSpace α where
  IsOpen u := ∀ ⦃d : Set α⦄, d ∈ D → d.Nonempty → DirectedOn (· ≤ ·) d → ∀ ⦃a : α⦄, IsLUB d a →
    a ∈ u → ∃ b ∈ d, Ici b ∩ d ⊆ u
  isOpen_univ := fun d _ ⟨b, hb⟩ _ _ _ _ ↦ ⟨b, hb, (Ici b ∩ d).subset_univ⟩
  isOpen_inter s t hs ht d hd₁ hd₂ hd₄ a hd₃ ha := by
    obtain ⟨b₁, hb₁d, hb₁ds⟩ := hs hd₁ hd₂ hd₄ hd₃ ha.1
    obtain ⟨b₂, hb₂d, hb₂dt⟩ := ht hd₁ hd₂ hd₄ hd₃ ha.2
    obtain ⟨c, hcd, hc⟩ := hd₄ b₁ hb₁d b₂ hb₂d
    exact ⟨c, hcd, fun e ⟨hce, hed⟩ ↦ ⟨hb₁ds ⟨hc.1.trans hce, hed⟩, hb₂dt ⟨hc.2.trans hce, hed⟩⟩⟩
  isOpen_sUnion := fun s h d hd₁ hd₂ hd₄ a hd₃ ⟨s₀, hs₀s, has₀⟩ ↦ by
    obtain ⟨b, hbd, hbds₀⟩ := h s₀ hs₀s hd₁ hd₂ hd₄ hd₃ has₀
    exact ⟨b, hbd, Set.subset_sUnion_of_subset s s₀ hbds₀ hs₀s⟩
```
and similarly:
```
def scottOn (α : Type*) [Preorder α] (D : Set (Set α)) : TopologicalSpace α := upperSet α ⊔ scottHausdorffOn α D
```
and then show that `ωScottContinuous` corresponds to continuity on this topology for `D = Set.range fun c : Chain α => Set.range c`. Then results like `OmegaCompletePartialOrder.continuous_id`, `OmegaCompletePartialOrder.continuous_comp` and `OmegaCompletePartialOrder.continuous_const` would just become the usual topological results. Furthermore we could subsume `Topology/OmegaCompletePartialOrder`.

This is probably two or three follow up PRs, so I didn't want to start tugging at that thread yet.

```

Reply:
Okay sure, but unless I've overlooked something you can already replace `OmegaCompletePartialOrder.Continuous` with `ωScottContinuous` everywhere without having to change the topology. Just use the fact that you've proved `OmegaCompletePartialOrder.Continuous f ↔ ωScottContinuous f`.

Reply:
Sorry, but I can't see how to compose `continuous'_iff_ωScottContinuous.mpr` with `ωScottContinuous f`.

```
variable {f : α → β}

#check continuous'_iff_ωScottContinuous.mpr (ωScottContinuous f)
```
Gives me:
```
application type mismatch
  continuous'_iff_ωScottContinuous.mpr (ωScottContinuous f)
argument
  ωScottContinuous f
has type
  Prop : Type
but is expected to have type
  ωScottContinuous ?m.27300 : Prop
```

Reply:
In any case, I still think the effort would be better expended on sorting this out properly in a follow up PR.

Reply:
You took my advice too literally 😉

I mean that now that you've proved that the two predicates are equivalent in lemma `continuous'_iff_ωScottContinuous`, you can replace `continuous'` in lemma statements by `ωScottContinuous`. All you need to do is prepend the proofs of the lemmas with `simp_rw [continuous'_iff_ωScottContinuous]`

Reply:
Nearly there in the latest commit, but can't see how to do `notBelow_isOpen`. Going to have to sleep now.

Reply:
> but can't see how to do `notBelow_isOpen`

I've sorted this now, after a night's sleep :)

I've now removed `Continuous'`.